### PR TITLE
Talking medbots

### PR DESCRIPTION
**Pull request by Software-Pirate-Captain**
Hello there!

This is a fairly small pull request that doesn't really change anything major to do with game balance, although I'll clarify up front that it does make medbots take a few extra seconds to start actually treating people if their speakers are enabled.

This adds a new folder full of sounds, some of which are then played by station-built medbots when treating people. There's sounds for being built, being destroyed, being emagged, types of damage, beating severity, things like that. Might update it with a few more if it's popular/well received!

I'm not aware of any changes to balance except for slightly delaying the actual healing injections, it's just a fun little fluff thing.

Some demos! Afraid there's a slight audio desynch, nothing I can do. However, they make the point well enough!

https://a.pomf.cat/mytyjy.webm
https://a.pomf.cat/isgdyb.webm
https://a.pomf.cat/ivfqpf.webm
https://a.pomf.cat/undpno.webm

### CODE REVIEW

**Comment by Kammerjunk** [comment]
>This is a fairly small pull request
>Files changed 20

¯\\_(ツ)_/¯

**Comment by Software-Pirate-Captain** [comment]
Those are just audio snippets, not really anything code-related. I mean, you've got a point, but it's not like I've entirely rewritten cult or something.

**Comment by yclatious** [comment]
Webm with sounds when

**Comment by Riekland** [comment]
I really like this
I thought I would hate it but I love it
Aren't there some unused lines for radiation and shit? That would be nice to have so that people coming into medbay could have the little boxes scream at them about the green glowing

**Comment by Software-Pirate-Captain** [comment]
Yeah, there's a lot, including for internal bleeding and broken bones, but I'd rather put up this small-ish version first and upgrade it later than spend forever struggling with a few tough parts, refusing to release the bits that do work.

Is that fair?

**Comment by Riekland** [comment]
No, you have to add a DNM tag and then leave the pr up for a few years while you make other shit and then come back to it after someone's added OC soundfiles and start drama
I don't even know how it works but I would have assumed something like IB is an on off thing in the limb that the medbot could check
Don't look at me though

**Comment by Software-Pirate-Captain** [comment]
I do intend to tackle it by cycling through every limb on the patient, as you suggest, but right now I'm trying to do some kind of managed, staged delivery rather than just throw the entire kitchen sink in one go and get out of my depth. :)

**Comment by Kammerjunk** [comment]
Sure, nothing wrong with deploying small bits fast rather than big bits slow far as I see

**Comment by TheFriggenSawbones** [comment]
It's neat but it could use some work maybe.

For one, they seem to overlap alerts sometimes which causes the sounds to overlap, its kinda messy.

This becomes more apparent when you have 2-3 (the norm I guess) medbots running at once.

I'm also not a fan of the flat-line sound, especially on medibot death.

**Comment by Software-Pirate-Captain** [comment]
I'm aware of the overlap but I've been struggling for a while on fixing it, since they don't always seem to pause when they're supposed to. Provided that they don't detect the wounded patient when they're right to it and have to actually move to it, there's no problem.

I'm not so sure about 2/3 bots being the norm, I normally only ever see one of them in medical and even then that's if you're lucky, but if you're sure I'll take your word for it.

I'll take another look at this if those things you mentioned become an issue.

**Comment by Riekland** [comment]
`>they don't always seem to pause when they're supposed to.`
I bet you'd need to redo all of bot code to fix that
Do we have a volunteer here or are you gonna be a LIL BITHC

**Comment by Exxion** [comment]
>tfw bots still aren't even mobs

**Comment by Software-Pirate-Captain** [comment]
Latter, sadly. I can lay out _exactly_ why they overlap and don't always report if you'd like, it's to do with the specialist reagent thing.

Essentially, if it detects that a person already has the reagent it would give them to fix a given damage type, it'll just move on. When it gives tricord for three of those, it means that as soon as it treats one, it'll ignore the others (and their associated voice lines), since they already have the "treatment" in them for anything that isn't it's special reagent.

You're right, it WOULD require almost entirely redoing medbots, no I'm nowhere near good enough to do that right now.


Not without tearing out the specialism thing which is actually a really nice piece of work.

**Comment by Shadowmech88** [comment]
@Software-Pirate-Captain 
If you wish to prevent the sounds from overlapping, you can have medbots play their alert sounds on a specific channel just for them. Only one sound can be played on a channel at a time, so you would only hear the last to be played rather than hearing all of them at once.

**Comment by Software-Pirate-Captain** [comment]
...Learn something new everyday. Is there some example code for this? I'll tweak it tomorrow and adjust the pullreq if so!

**Comment by Shadowmech88** [comment]
You just need to add a new sound channel define, as seen here: 
https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/__DEFINES/setup.dm#L1372

Then give that channel define as an argument to `playsound()`, for example 
`playsound(src.loc, 'sound/medbot/Chemical_Detected.ogg', 35, channel = CHANNEL_MEDBOT)`

**Comment by Software-Pirate-Captain** [comment]
Implemented channels as suggested. Doesn't seem to be queuing them correctly, instead dropping one that would overlap instead of playing both at once, but that's still a big improvement.

Anything else?

**Comment by Shadowmech88** [comment]
>Doesn't seem to be queuing them correctly, instead dropping one that would overlap instead of playing both at once

That's intended. When using a single channel, playing a sound terminates any sounds that were already playing.
I didn't mean to imply that they would be queued and played one after another.

**Comment by Software-Pirate-Captain** [comment]
Then I guess that's working as intended! :)

With that sorted, I think that's everything I'd wanted to tackle for this one. Is the next step to sit back and wait for it to be merged or closed?

**Comment by ghost** [comment]
Wait five years for collabs to do their job

**Comment by ihadtoregisterforthis** [comment]
Could you provide a demonstration of this in action?

**Comment by ComicIronic** [comment]
This is still waiting for a video with sound.

**Comment by Software-Pirate-Captain** [comment]
Got a somewhat busy IRL schedule right now, I'll get on it as soon as a gap in finals opens up. Apologies.

**Comment by Software-Pirate-Captain** [comment]
Aaaaaalrighty, got some Webms with sound for you all. How would you like them hosted? I'll throw them up on a burner Mega account for now.

**Comment by Shadowmech88** [comment]
Puu.sh is the simplest way to host, I think.

**Comment by Software-Pirate-Captain** [comment]
There's gotta be a way that doesn't involve another piece of dodgy freeware?

Also, turns out recording at a high framerate/resolution desynchs the audio, need to reshoot them.

**Comment by Shadowmech88** [comment]
>dodgy freeware

I thought it was fairly common and dependable.

Anyway, there are plenty of free direct-access filehosting sites you can upload to from a browser. https://pomf.cat/ for example.

**Comment by Exxion** [comment]
It is fairly common, though at one point someone did manage to hijack their update server or something and give a bunch of people viruses
They're probably more careful now though

**Comment by Software-Pirate-Captain** [comment]
I'll use the Pomfcat one. As I said, turns out I need to reshoot some of the stuff due to an audio desynch. Please bear with me!

**Comment by Software-Pirate-Captain** [comment]
https://a.pomf.cat/mytyjy.webm
https://a.pomf.cat/isgdyb.webm
https://a.pomf.cat/ivfqpf.webm
https://a.pomf.cat/undpno.webm

Ta-Da!~

Apologies for the desynch, I've tried a couple of things but can't fix it. The software seems to just be behind what going on by half a second. Seems to get worse as time goes on, since towards the end of the long one it's several seconds out of whack, but I don't think there's anything I can do about it.

Missed the one for it being emagged, but it's just another voice clip.

**Comment by Shadowmech88** [comment]
Put them in the OP as well.

**Comment by Software-Pirate-Captain** [comment]
Done. What're your thoughts?

**Comment by Shadowmech88** [comment]
I think it sounds fine.

**Comment by yclatious** [comment]
Sounds good and fits surprisingly well so, good stuff.

**Comment by Software-Pirate-Captain** [comment]
Alright. Any ETA on it being merged, or is it just the waiting game now?

File: code/game/machinery/bots/medbot.dm [hunk 1]
```diff
@@ -436,30 +439,55 @@
...
+				say("Antitoxin administered!")
+				sleep(25)
+
 
 	if(!reagent_id) //If they don't need any of that they're probably cured!
 		oldpatient = patient
 		patient = null
 		currently_healing = 0
 		last_found = world.time
-		var/message = pick("All patched up!","An apple a day keeps me away.","Feel better soon!")
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm:461]: feature loss!

**Comment by ghost** [comment]
>there's a radiation detected.ogg

You've given me an idea

**Comment by Software-Pirate-Captain** [comment]
I pulled those messages out on purpose due to finding them annoying as hell, guess I shouldn't have just assumed that everyone else did too.

As for the unusued lines, I intend to come back to this when the real world quietens down for me and do something with them. The internal bleeding, major and minor fracture lines as well as the radiation one are just perfect.

**Comment by Software-Pirate-Captain** [comment]
Any updates on this one?  It's just sitting around being all open right now, and I'm not aware of any action I need to take?

**Comment by SonixApache** [comment]
HL1 bitching betty sounds are nice, but it feels wrong having them on something that isn't a suit

**Comment by ghost** [comment]
holy moly, this's been open a month.

Have one last scour through the code and find places where it doesn't check for shut up when it comes to making the sounds.

Heck, I'll go through it now

File: code/game/machinery/bots/medbot.dm [hunk 2]
```diff
@@ -436,30 +439,55 @@
 	if (!reagent_id && (virus))
 		if(!C.reagents.has_reagent(treatment_virus))
 			reagent_id = treatment_virus
+			playsound(src.loc, 'sound/medbot/Biohazard_detected.ogg', 35, channel = CHANNEL_MEDBOTS)
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm:442]: have a shut_up check for here

File: code/game/machinery/bots/medbot.dm [hunk 3]
```diff
@@ -414,8 +417,8 @@
 		return
 
 	if(C.stat == 2)
-		var/death_message = pick("No! NO!","Live, damnit! LIVE!","I...I've never lost a patient before. Not today, I mean.")
-		speak(death_message)
+		playsound(src.loc, 'sound/medbot/Flatline_custom.ogg', 35, channel = CHANNEL_MEDBOTS)
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm:420]: shut_up check needed here

**Review by ghost — CHANGES_REQUESTED** [review]
just have more of the sounds adhere to shut_up and it'll be golden

**Pull request merged**